src/Treeview.ts

```diff
@@ -297,6 +292,40 @@ export class JsonOutlineProvider implements vscode.TreeDataProvider<string> {
         }
         return;
     }
+
+    private updateTreeState() {
+        const activeEditor: vscode.TextEditor = vscode.window.activeTextEditor;
+        const document: vscode.TextDocument = !!activeEditor ? activeEditor.document : null;
+        this.parseTree(document);
+        const showTreeView = this.isArmTemplate(document);
+
+        if (showTreeView) {
+            this.refresh();
+        }
+
+        this.setTreeViewContext(showTreeView);
+    }
+
+    private isArmTemplate(document?: vscode.TextDocument): boolean {
+        return !!document && document.languageId.toLowerCase() === 'json' && Utilities.isValidSchemaUri(this.getSchemaUri());
+    }
+
+    private setTreeViewContext(visible: boolean) {
+        vscode.commands.executeCommand('setContext', 'showArmJsonView', visible);
+    }
+
+    private getSchemaUri(): string {
+        if (!!this.tree) {
```

Comment:
Can any of these checks throw an exception? For example, `Json.asObjectValue` or `Json.asStringValue` or `schema.toString()`?

If it's remotely possible, I'd prefer to wrap this in a try/catch and ignore all errors since we perform this check for every single json file (correct?)

Reply:
No, these functions shouldn't throw exceptions. If the values can't be converted, they return null, which code does the check.